lib/content-repo.js

```diff
@@ -13,14 +13,23 @@ module.exports = {
      * @param {Number} height The desired height of the thumbnail.
      * @param {string} resizeMethod The thumbnail resize method to use, either
      * "crop" or "scale".
+     * @param {Boolean} allowDirectLinks If true, return any non-mxc URLs
+     * directly. Fetching such URLs will leak information about the user to
+     * anyone they share a room with. If false, will return the emptry string
+     * for such URLs.
      * @return {string} The complete URL to the content.
      */
-    getHttpUriForMxc: function(baseUrl, mxc, width, height, resizeMethod) {
+    getHttpUriForMxc: function(baseUrl, mxc, width, height,
+                               resizeMethod, allowDirectLinks) {
```

Comment:
I feel quite strongly that this should be the default: the behaviour should be secure by default and clients can disable the security features if they want to. It's a breaking change, but IMO it's preferable to leave clients a bit broken but secure than keep the old, insecure behaviour.